Google.Common/src/Google.Common.Embeddable/Preconditions.cs

```diff
@@ -107,5 +107,54 @@ internal static void CheckState<T>(bool condition, string format, T arg0)
                 throw new InvalidOperationException(string.Format(format, arg0, arg1));
             }
         }
+
+
+        /// <summary>
+        /// Checks that given argument-based condition is met, throwing an <see cref="ArgumentException"/> otherwise.
+        /// </summary>
+        /// <param name="condition">The condition to test.</param>
```

Comment:
This is not really the condition but rather the result of the condition, I would state as much in the comment. As the comment stands it seems to imply, at least to me, that it accepts a Func<bool>, that is, the condition itself.